server/plugins/client/index.js

```diff
@@ -19,6 +19,29 @@ function register (server, options, next) {
   // TODO: add /hoodie/client.min.js path
   // https://github.com/hoodiehq/hoodie-client/issues/34
 
+  var decideAction = function (hasUpdate, bundleBuffer) {
+    if (!hasUpdate) {
+      log.info('client', 'bundle is up to date')
+      return
+    }
+
+    if (options.config.inMemory) {
+      log.silly('client', 'running in memory, not writing bundle to ' + bundleTargetPath)
+      return
+    }
+
+    log.info('client', 'bundle is out of date')
+    log.silly('client', 'writing bundle to ' + bundleTargetPath)
+    fs.writeFile(bundleTargetPath, bundleBuffer, function (error) {
+      if (error) {
+        log.warn('client', 'could not write to ' + bundleTargetPath + '. Bundle cannot be cached and will be re-generated on server restart')
+        return
+      }
+
+      log.info('client', 'bundle written to ' + bundleTargetPath)
+    })
+  }
```

Comment:
I would suggest we add a `inMemory` and a `bundleTargetPath` argument, so that the function no longer access state outside of its scope. Then move it to the very end of the file, outside of the `register` function. And please define it with `function decideAction (hasUpdate, bundleBuffer) {` instead of `var decideAction = function (hasUpdate, bundleBuffer) {`